webapp/src/auth.ts

```diff
@@ -256,6 +263,43 @@ async function fetchUserAsync() {
     }
 }
 
+
+export async function updateUserPreferencesAsync(newPref: Partial<UserPreferences>) {
```

Comment:
>export [](start = 0, length = 6)

Code organization: Put exported functions above the "Private functions" section.